tuxemon/core/states/world/worldstate.py

```diff
@@ -617,14 +617,11 @@ def get_exits(self, position, collision_map=None, skip_nodes=None):
         if skip_nodes is None:
             skip_nodes = set()
 
-        # if there are explicit way to exit this position use that
-        # information only and do not check surrounding tiles.
+        # if there are explicit way to exit this position use that information,
         # handles 'continue' and 'exits'
         tile_data = collision_map.get(position)
         if tile_data:
             exits = self.get_explicit_tile_exits(position, tile_data, skip_nodes)
-            if exits:
```

Comment:
why is this changed?

Reply:
I moved it below in the `for direction, neighbor` loop. This is what allowed the player to walk into other collisions from a neighboring collision defining an `exit_to` toward it: If exits were found here, they were immediately returned as a valid position. With my change, if exits are defined, we instead ensure that the target position is present in them... if not then it's invalid, if yes then it's valid but we now also proceed with the other checks to make sure the position can be walked to.